### PR TITLE
Environment properties with dots

### PR DESCRIPTION
**Pull request by worldtiki**
### :pencil: Description
Environment properties with periods are not passed to pitchfork.
This was the case for Alpine for some time but it looks like it's also happening for Debian now.
There are several tickets with more details about this issue (e.g.: https://github.com/docker-library/openjdk/issues/135) but in short the alternatives are:
- promote the usage of POSIX names (underscore + uppercase)
- use bash

I'm opening this pr to get some feedback on the pros and cons of both.

### CODE REVIEW

File: docs/about/getting-started.md [hunk 1]
```diff
@@ -11,24 +11,24 @@ HTTP is the default ingress and is always enabled. The following properties can
```
Comment (worldtiki) [inline docs/about/getting-started.md]: this is option 1: use underscore + uppercase

File: Dockerfile [hunk 2]
```diff
@@ -36,17 +36,11 @@ ENV JAVA_HOME=/opt/jdk-mini
 ENV PATH="$PATH:$JAVA_HOME/bin"
 ENV DIRPATH /pitchfork
 
-
 # Create some dirs and copy pitchfork jar
 RUN mkdir -p $DIRPATH
 COPY target/pitchfork.jar $DIRPATH/
 
 RUN chmod 755 $DIRPATH/pitchfork.jar
 WORKDIR $DIRPATH
```
Comment (worldtiki) [inline Dockerfile]: this is option 2: bash

File: Dockerfile [hunk 3]
```diff
@@ -22,7 +22,7 @@ ENV PATH="$PATH:$JAVA_HOME/bin"
 RUN jlink \
      --module-path /opt/java/jmods \
      --compress=2 \
-     --add-modules jdk.jfr,jdk.management.agent,java.base,java.logging,java.xml,jdk.unsupported,java.sql,java.naming,java.desktop,java.management,java.security.jgss,java.instrument \
```
Comment (worldtiki) [inline Dockerfile]: this is unrelated, just ignore this line

**Comment by worldtiki** [comment]
fyi @NersesAM

File: docs/about/getting-started.md (hunk 1 above)
Comment (NersesAM) [reply docs/about/getting-started.md]: I would go with option 1, because it will always work

Comment (worldtiki) [reply docs/about/getting-started.md:11]: 18 days delay in replying :x sorry.
i agree. i'll prepare a pr soon with the changes.

File: docs/about/getting-started.md [hunk 4]
```diff
@@ -54,39 +54,39 @@ Kafka output is disabled by default. You can enable and configure it using the f
...
+| PITCHFORK_FORWARDERS_HAYSTACK_KINESIS_CLIENT_ENDPOINT_SIGNING_REGION_NAME | us-west-1                               | Used when kinesis client config-type is ENDPOINT. The region to use for signing of requests                                 |
+| PITCHFORK_FORWARDERS_HAYSTACK_KINESIS_AUTH_CONFIG_TYPE                    | DEFAULT                                 | Authentication method to use. Possible values are DEFAULT or BASIC                                                          |
+| PITCHFORK_FORWARDERS_HAYSTACK_KINESIS_AUTH_BASIC_AWS_ACCESS_KEY           | accesskey                               | Used when authentication-type is BASIC. The AWS access key                                                                  |
+| PITCHFORK_FORWARDERS_HAYSTACK_KINESIS_AUTH_basic_AWS_SECRET_KEY           | secretkey                               | Used when authentication-type is BASIC. The AWS secret access key                                                           |
```
Comment (nikos912000) [inline docs/about/getting-started.md:75]: `basic` is lowercase, can we convert to uppercase?

File: docs/about/getting-started.md [hunk 5]
```diff
@@ -113,34 +113,34 @@ By default Pitchfork exposes both a /metrics and a /prometheus endpoint (see sec
...
-This is useful if you have multiple instances of Pitchfork running and you would want your metrics to look like this: `pitchfork.instance-01.metric-name=123` 
+This is useful if you have multiple instances of Pitchfork running and you would want your metrics to look like this: `pitchfork.instance-01.metric-name=123`
 
 ```
-management.metrics.tags.application=pitchfork
-management.metrics.tags.instance=instance-01
-management.metrics.export.graphite.tags-as-prefix=application,instance
+MANAGEMENT_METRICS_MANAGEMENT_METRICS_TAGS_APP=pitchfork
+MANAGEMENT_METRICS_MANAGEMENT_METRICS_TAGS_INSTANCE=instance-01
+MANAGEMENT_METRICS_MANAGEMENT_METRICS_EXPORT_GRAPHITE_TAGS_AS_PREFIX=APP,INSTANCE
```
Comment (nikos912000) [inline docs/about/getting-started.md:127]:
```suggestion
MANAGEMENT_METRICS_TAGS_APP=pitchfork
MANAGEMENT_METRICS_TAGS_INSTANCE=instance-01
MANAGEMENT_METRICS_EXPORT_GRAPHITE_TAGS_AS_PREFIX=APP,INSTANCE
```

File: docs/about/getting-started.md (hunk 4 above)
Comment (nikos912000) [inline docs/about/getting-started.md:75]:
```suggestion
| PITCHFORK_FORWARDERS_HAYSTACK_KINESIS_AUTH_BASIC_AWS_SECRET_KEY           | secretkey                               | Used when authentication-type is BASIC. The AWS secret access key                                                           |
```

File: Dockerfile [hunk 6]
```diff
@@ -36,17 +36,11 @@ ENV JAVA_HOME=/opt/jdk-mini
...
 RUN chmod 755 $DIRPATH/pitchfork.jar
 WORKDIR $DIRPATH
 
-# Set timezone (for logs) and run pitchfork
-CMD export TZ=$(date +%Z) &&\
-    exec $JAVA_HOME/bin/java \
-         $JAVA_JVM_ARGS \
-         -jar \
-         pitchfork.jar
+CMD ["/bin/bash", "-c", "$JAVA_HOME/bin/java $JAVA_JVM_ARGS -jar pitchfork.jar"]
```
Comment (NersesAM) [inline Dockerfile:46]: will java be PID 1 with this? so that shutdown is graceful

File: Dockerfile [hunk 7]
```diff
@@ -36,17 +36,11 @@ ENV JAVA_HOME=/opt/jdk-mini
 ENV PATH="$PATH:$JAVA_HOME/bin"
```
Comment (NersesAM) [inline Dockerfile:36]: why you need this line if you are starting java process like $JAVA_HOME/bin/java

File: Dockerfile [hunk 8]
```diff
@@ -36,17 +36,11 @@ ENV JAVA_HOME=/opt/jdk-mini
...
 ENV DIRPATH /pitchfork
```
Comment (NersesAM) [inline Dockerfile:37]: this can be ARG so it's not used as env var

File: Dockerfile (hunk 6 above)
Comment (worldtiki) [reply Dockerfile:46]: i'm going to revert this change.
this was for option 2 (force using bash) that still allows properties with dots.

File: Dockerfile (hunk 8 above)
Comment (worldtiki) [reply Dockerfile:37]: yeap, i will change that

File: Dockerfile [hunk 9]
```diff
@@ -36,17 +36,11 @@ ENV JAVA_HOME=/opt/jdk-mini
...
 
-
 # Create some dirs and copy pitchfork jar
 RUN mkdir -p $DIRPATH
 COPY target/pitchfork.jar $DIRPATH/
 
 RUN chmod 755 $DIRPATH/pitchfork.jar
```
Comment (NersesAM) [inline Dockerfile:43]: is this necessary? I think it should work without this line

Comment (worldtiki) [reply Dockerfile]: Yeap, not needed. Removed. Thanks :)
